### DISCUSSION TITLE
npm run dev does not work, it gives error

### DISCUSSION THREAD

**Discussion by ShakilSayyed** [hbstack/.github#22] [Q&A] [answered]
Error: command error: failed to create config from modules config: unknown output format "searchindex" for kind "home"

**Comment by razonyang** [comment]
Hi @ShakilSayyed, it most likely that the [search](https://hugomods.com/en/docs/search/) module wasn't imported. The _full_ error log is more helpful to diagnose. BTW, the theme requires _extended_ Hugo `>= v0.111.3`, please check if it's related to the Hugo version (`hugo version`).

It would be also helpful if you have a public repo for reproducing this issue.

**Reply by ShakilSayyed** [reply]
Thanks @razonyang, I am using the latest versin of extended hugo that is hugo v0.114.0- and also updated go version from go version go1.20.1 windows/amd64 to go1.20.5.
I am not yet able to run the programe, I have followed the instruction given on the site and trying to run it as it is. Once it starts to run I may do some changes and upload the repo.
The very first time I ran `npm run dev`. the output is as follows:
git clone --depth 1 https://github.com/hbstack/theme blog      
Cloning into 'blog'...
remote: Enumerating objects: 210, done.
remote: Counting objects: 100% (210/210), done.
remote: Compressing objects: 100% (156/156), done.
remote: Total 210 (delta 15), reused 187 (delta 14), pack-reused 0
Receiving objects: 100% (210/210), 60.53 MiB | 1.11 MiB/s
Receiving objects: 100% (210/210), 61.07 MiB | 1.11 MiB/s, done.
Resolving deltas: 100% (15/15), done.
Updating files: 100% (202/202), done.
cd blog/                                                    
git st                      
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean


added 91 packages, and audited 92 packages in 4s

21 packages are looking for funding
  run `npm fund` for details

1 high severity vulnerability

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
npm run dev                

> dev
> hugo server --gc --disableFastRender --enableGitInfo -D
hugo: downloading modules …
hugo: collected modules in 4125 ms
Hugo provides its own webserver which builds and serves the site.
While hugo server is high performance, it is a webserver with limited options.

'hugo server' will avoid writing the rendered and served content to disk,
preferring to store it in memory.

By default hugo will also watch your files for any changes you make and
automatically rebuild the site. It will then live reload any open browser pages
and push the latest content to them. As most Hugo sites are built in a fraction
of a second, you will be able to save and see your changes nearly instantly.

Usage:
  hugo server [command] [flags]
  hugo server [command]

Aliases:
  server, serve

Available Commands:
  trust       Install the local CA in the system trust store.


Use "hugo server [command] --help" for more information about a command.

Error: command error: failed to create config from modules config: unknown output format "searchindex" for kind "home"

**Reply by razonyang** [reply]
Hi, do you use WSL or native Windows (PowerShell)? I'm trying to reproduce it.

**Reply by ShakilSayyed** [reply]
I was using gitbash in windows terminal, also tried to run it in powershell but the result is same.

**Reply by ShakilSayyed** [reply]
I am not using WSL at all

**Reply by razonyang** [reply, marked as answer]
@ShakilSayyed Hmm, I can't reproduce it, the instructions work as expected on both of WSL and PowerShell.
According to your logs, there are some extra output of Hugo CLI, how do you install the Hugo? Please give the official binary (download from [releases](https://github.com/gohugoio/hugo/releases) which named `extended` and `windows amd64`) a try.

Use the downloaded Hugo CLI directly instead of `npm run dev` for testing.

```sh
D:\Path\to\hugo.exe server --gc --disableFastRender --enableGitInfo -D
```

![image](https://github.com/hbstack/.github/assets/17720932/c6818336-a694-45fe-a0b8-077130667469)

![image](https://github.com/hbstack/.github/assets/17720932/026d3007-9237-4985-857f-a4112ddbfcc9)

**Reply by ShakilSayyed** [reply]
Thanks again, I tried on my other machine under same environment and it is working. I will figure out what is wrong with the first one and update here.

**Reply by razonyang** [reply]
My pleasure. I'm closing this as answered.
